### PR TITLE
feat/app context

### PR DESCRIPTION
**Pull request by sedghi**
- **Remove unused storeContexts from commandsModule**
- **Refactor code and fix linting issues**
- **clean up mode tsx**
- **Add AppContextService and update contextProviders**
- **Refactor toolbar and remove unused code**
- **Refactor toolbar buttons and add tooltips**
- **Update Toolbar and ViewportGrid components**
- **Refactor UI and contextProviders modules**
- **Add disabled state to toolbar buttons and side panel tabs**
- **Refactor ViewerLayout and App components**
- **Refactor toolbar button handling and add display set condition**
- **Refactor toolbar button handling logic**
- **Add button condition for displaySets.isReconstructable*** ***Add button condition for toolGroup.id !== 'mpr'*** ***Update ListMenu component to handle disabled items*** ***Add evaluateToolbarButtonState module**
- **Refactor toolbar button evaluation and update list menu theme**
- **Remove isActive property from moreTools and toolbarButtons**
- **Refactor toolbar button state handling**
- **Refactor toolbar button toggling logic**
- **Refactor Toolbar component and ToolbarService**
- **Add SplitButtonToolbar component**
- **clean up**

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] OS: 
- [] Node version: 
- [] Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: extensions/cornerstone/src/init.tsx [hunk 1]
```diff
@@ -248,19 +248,19 @@ export default async function init({
 
     const activeTools = toolbarService.getActiveTools();
 
-    activeTools.forEach(tool => {
-      if (!toolGroup?._toolInstances?.[tool]) {
+    activeTools.forEach(toolId => {
+      if (!toolGroup?._toolInstances?.[toolId]) {
```
Comment (wayfarer3130) [inline extensions/cornerstone/src/init.tsx:252]: Please access via some sort of getter - retrieving the tool directly makes it hard to change the representation in the future.

File: platform/core/src/services/ToolBarService/ToolbarService.ts [hunk 2]
```diff
@@ -2,35 +2,43 @@ import merge from 'lodash.merge';
...
-  secondary?: Button;
-  items?: Button[];
+enum ButtonInteractionType {
+  ACTION = 'action',
+  TOGGLE = 'toggle',
+  TOOL = 'tool',
 }
 
-export interface Button extends Commands {
+export interface ButtonProps {
```
Comment (wayfarer3130) [inline platform/core/src/services/ToolBarService/ToolbarService.ts:18]: Can you change this to a type please?  That works better for extending it with customizations and other info.

File: platform/core/src/services/ToolBarService/ToolbarService.ts [hunk 3]
```diff
@@ -2,35 +2,43 @@ import merge from 'lodash.merge';
...
+export interface NestedButtonProps {
+  groupId: string;
+  items: ButtonProps[];
+  primary: ButtonProps;
+  secondary: ButtonProps;
+}
+
+export interface Button {
+  id: string;
+  props: ButtonProps | NestedButtonProps;
```
Comment (wayfarer3130) [inline platform/core/src/services/ToolBarService/ToolbarService.ts:38]: If this is going to change so much, can we make the button just contain it's props directly, and use two type includes?

File: platform/core/src/services/ToolBarService/ToolbarService.ts [hunk 4]
```diff
@@ -135,48 +152,48 @@ export default class ToolbarService extends PubSubService {
...
     }
 
+    // if not interaction type, assume the type can be used
+    const { interactionType = type } = interaction;
+
+    const prevPrimaryToolId = this.state.primaryToolId;
+
     switch (interactionType) {
-      case 'action': {
+      case ButtonInteractionType.ACTION: {
```
Comment (wayfarer3130) [inline platform/core/src/services/ToolBarService/ToolbarService.ts:171]: Could we get rid of button interaction type, and require toggle buttons to implement a callback that gets the current button state/class set?  That way changing the viewport or camera or whatever has listeners can trigger toggle button updates as required, and if the action fails to trigger, it won't get set as being toggled.

**Comment by sedghi** [comment]
Not ready yet

**Pull request closed without merging**